Artifacts not uploaded if tests fail

From a recent test run we can see the uploading of artifacts was skipped, presumably due to the failure:

```
2022-05-09T17:36:10.3834070Z --- FAIL: TestQuick (76.32s)
2022-05-09T17:36:10.3834510Z     --- FAIL: TestQuick/Ensure_pod_and_worker_logs_gathered (0.07s)
2022-05-09T17:36:10.3834922Z FAIL
2022-05-09T17:36:10.3846856Z FAIL	github.com/vmware-tanzu/sonobuoy/test/integration	76.333s
2022-05-09T17:36:10.3888553Z FAIL
2022-05-09T17:36:11.1565132Z ##[error]Process completed with exit code 1.
2022-05-09T17:36:11.1637546Z Post job cleanup.
2022-05-09T17:36:11.3292779Z [command]/usr/bin/git version
2022-05-09T17:36:11.3348947Z git version 2.36.0
2022-05-09T17:36:11.3406180Z Temporarily overriding HOME='/home/runner/work/_temp/faaa8c4d-53a5-4b09-bd2b-83401b4b4ae8' before making global git config changes
2022-05-09T17:36:11.3406952Z Adding repository directory to the temporary git global config as a safe directory
2022-05-09T17:36:11.3410977Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/sonobuoy/sonobuoy
2022-05-09T17:36:11.3464967Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2022-05-09T17:36:11.3499724Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-05-09T17:36:11.3955885Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-05-09T17:36:11.3992031Z http.https://github.com/.extraheader
2022-05-09T17:36:11.4002824Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2022-05-09T17:36:11.4042140Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-05-09T17:36:11.4998106Z Cleaning up orphan processes
```

Just need to add `if: always` to the step/job.

--- a/.github/workflows/ci-test.yaml
+++ b/.github/workflows/ci-test.yaml
@@ -129,6 +129,7 @@ jobs:
           load_test_images_into_cluster
           VERBOSE=true SONOBUOY_CLI=../../build/linux/amd64/sonobuoy integration
       - name: Save artifacts
+        if: ${{ always() }}
         uses: actions/upload-artifact@v2
         with:
           name: sonobuoy-test-archives-${{ github.run_id }}

